File: laocai-bot-spring-boot-starter/src/main/java/indi/dkx/laocai/bot/model/segment/IncomingReplySegment.java
package indi.dkx.laocai.bot.model.segment;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.List;

/**
 * 入站的引用回复消息段。
 * <p>
 * 只在反序列化时产生，携带被回复消息的完整内容。
 */
public final class IncomingReplySegment implements Segment {

    record Data(Long messageSeq, Long senderId, String senderName, Long time, List<Segment> segments) {
        IncomingReplySegment toSegment() {
            return new IncomingReplySegment(this);
        }
    }

    private final Data data;

    private IncomingReplySegment(Data data) {
        this.data = data;
    }

    @Override
    public String getType() {
        return "reply";
    }

    @Override
    public Data getData() {
        return data;
    }

    @JsonIgnore
    public Long getMessageSeq() {
        return data.messageSeq();
    }

    @JsonIgnore
    public Long getSenderId() {
        return data.senderId();
    }

    @JsonIgnore
    public String getSenderName() {
        return data.senderName();
    }

    @JsonIgnore
    public Long getTime() {
        return data.time();
    }

    @JsonIgnore
    public List<Segment> getSegments() {
        return data.segments();
    }
}

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/segment/IncomingReplySegment.kt
package indi.kyson.laocai.bot.segment

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * 入站的引用回复消息段。
 *
 * 只在反序列化时产生，携带被回复消息的完整内容。
 */
class IncomingReplySegment private constructor(override val data: Data) : Segment {

    data class Data(
        @JsonProperty("message_seq")
        val messageSeq: Long,
        @JsonProperty("sender_id")
        val senderId: Long,
        @JsonProperty("sender_name")
        val senderName: String?,
        val time: Long,
        val segments: List<Segment>,
    ) {
        fun toSegment(): IncomingReplySegment = IncomingReplySegment(this)
    }

    override val type: String
        get() = "reply"

    @get:JsonIgnore
    val messageSeq: Long
        get() = data.messageSeq

    @get:JsonIgnore
    val senderId: Long
        get() = data.senderId

    @get:JsonIgnore
    val senderName: String?
        get() = data.senderName

    @get:JsonIgnore
    val time: Long
        get() = data.time

    @get:JsonIgnore
    val segments: List<Segment>
        get() = data.segments
}
